search_service/proxy/elasticsearch.py

```diff
@@ -377,6 +376,130 @@ def fetch_table_search_results(self, *,
                                    query_name=query_name,
                                    model=Table)
 
+    @staticmethod
+    def convert_query_json_to_query_dsl(search_request: dict,
+                                        query_term: str) -> str:
+        """
+        Convert the generic query json to query DSL
+        e.g
+        ```
+        {
+            'type': 'AND'
+            'filters': {
+                'database': ['hive', 'bigquery'],
+                'schema': ['test-schema1', 'test-schema2'],
+                'table': ['*amundsen*'],
+                'column': ['*ds*']
+                'tag': ['test-tag']
+            }
+        }
+
+        This generic JSON will convert into DSL depending on the backend engines.
+
+        E.g in Elasticsearch, it will become
+        'database':('hive' OR 'bigquery') AND
+        'schema':('test-schema1' OR 'test-schema2') AND
+        'table':('*amundsen*') AND
+        'column':('*ds*') AND
+        'tag':('test-tag')
+        ```
+
+        :param search_request:
+        :return: The search engine query DSL
+        """
+        filter_list = search_request.get('filters')
+        add_query = ''
+        query_dsl = ''
+
+        if filter_list:
+            query_dsl = ' AND '.join([(category + ':' + '(' +
+                                     ' OR '.join(item_list) + ')')
+                                      for category, item_list in filter_list.items()])
+
+        # exact match
+        for k, v in TABLE_MAPPING.items():
+            query_dsl = query_dsl.replace(k, v)
+
+        if query_term:
+            if '.' in query_term:
+                # query_term is schema.table
+                schema, table = query_term.split('.')
```

Comment:
this will error if there is more than one period. I wonder if there is a case where a user might search something with more than one. It seems possible -- I think this should be handled differently so that it won't cause an error based on too many periods

Reply:
Also, I'm curious about handling the case where the user provides a single period like ".*" (which I would expect to return all results)

Reply:
Will let Tao comment on this one as well. At Lyft a lot of our users do explicitly search for `schema.table` and I considered perhaps this logic might not work well for everyone.

Also, advanced search syntax is no longer supposed to be entered in the query term -- we want users to use the filters for refining their search. Perhaps this means the frontend needs some logic to prevent "*" in the query term.

Reply:
I bring up multiple periods also because we have multiple periods in some of our cluster names and I can see this being weird

Reply:
Yeah essentially I think this was some experimental code we had in our internal code base for testing and this specific logic would yield good results and Lyft but not for other.  I'm thinking we will either remove this from the default implementation and/or structure things on such a way that folks can override/extend the generation logic of `query_dsl` and `add_query`